packages/beacon-node/src/chain/rewards/blockRewards.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  CachedBeaconStateAllForks,
+  CachedBeaconStateAltair,
+  CachedBeaconStatePhase0,
+  getAttesterSlashableIndices,
+  processAttestationsAltair,
+} from "@lodestar/state-transition";
+import {allForks, altair, phase0} from "@lodestar/types";
+import {ForkName, WHISTLEBLOWER_REWARD_QUOTIENT} from "@lodestar/params";
+import {routes} from "@lodestar/api";
+
+export type BlockRewards = routes.beacon.BlockRewards;
+type SubRewardValue = number; // All reward values should be integer
+
+/**
+ * Calculate total proposer block rewards given block and the beacon state of the same slot before the block is applied (preState)
+ * postState can be passed in to read reward cache if available
+ * Standard (Non MEV) rewards for proposing a block consists of:
+ *  1) Including attestations from (beacon) committee
+ *  2) Including attestations from sync committee
+ *  3) Reporting slashable behaviours from proposer and attester
+ */
+export async function computeBlockRewards(
+  block: allForks.BeaconBlock,
+  preState: CachedBeaconStateAllForks,
+  postState?: CachedBeaconStateAllForks
+): Promise<BlockRewards> {
+  const fork = preState.config.getForkName(block.slot);
+  const {attestations: cachedAttestationsReward = 0, syncAggregate: cachedSyncAggregateReward = 0} =
+    postState?.proposerRewards || {};
+  let blockAttestationReward = cachedAttestationsReward;
+  let syncAggregateReward = cachedSyncAggregateReward;
+
+  if (blockAttestationReward === 0) {
+    blockAttestationReward =
+      fork === ForkName.phase0
+        ? computeBlockAttestationRewardPhase0(block as phase0.BeaconBlock, preState as CachedBeaconStatePhase0)
+        : computeBlockAttestationRewardAltair(block as altair.BeaconBlock, preState as CachedBeaconStateAltair);
+  }
+
+  if (syncAggregateReward === 0) {
+    syncAggregateReward = computeSyncAggregateReward(block as altair.BeaconBlock, preState as CachedBeaconStateAltair);
+  }
+
+  const blockProposerSlashingReward = computeBlockProposerSlashingReward(block, preState);
+  const blockAttesterSlashingReward = computeBlockAttesterSlashingReward(block, preState);
+
+  const total =
+    blockAttestationReward + syncAggregateReward + blockProposerSlashingReward + blockAttesterSlashingReward;
+
+  return {
+    proposerIndex: block.proposerIndex,
+    total,
+    attestations: blockAttestationReward,
+    syncAggregate: syncAggregateReward,
+    proposerSlashings: blockProposerSlashingReward,
+    attesterSlashings: blockAttesterSlashingReward,
+  };
+}
+
+/**
+ * TODO: Calculate rewards received by block proposer for including attestations.
+ */
+function computeBlockAttestationRewardPhase0(
+  _block: phase0.BeaconBlock,
+  _preState: CachedBeaconStatePhase0
+): SubRewardValue {
+  throw new Error("Unsupported fork! Block attestation reward calculation is not yet available in phase0");
```

Comment:
```suggestion
  throw new Error("Unsupported fork! Block attestation reward calculation is not available in phase0");
```

tbh not practically useful, no need to get folks' hopes up XD